kernel/cpu/src/aarch64.rs

```diff
@@ -2,32 +2,61 @@
 
 use cortex_a::registers::MPIDR_EL1;
 use tock_registers::interfaces::Readable;
-
+use derive_more::{Display, Binary, Octal, LowerHex, UpperHex};
+use irq_safety::RwLockIrqSafe;
 use core::fmt;
+use alloc::vec::Vec;
+
 use super::CpuId;
-use derive_more::{Display, Binary, Octal, LowerHex, UpperHex};
+
+// The vector of CpuIds for known and online CPU cores
+static ONLINE_CORES: RwLockIrqSafe<Vec<CpuId>> = RwLockIrqSafe::new(Vec::new());
```

Comment:
nit: for consistency, we're migrating terms to `cpu` instead of `core`
```suggestion
static ONLINE_CPUS: RwLockIrqSafe<Vec<CpuId>> = RwLockIrqSafe::new(Vec::new());
```

Reply:
do we need a `RwLock` for this? If it's not absolutely required, we should use `Mutex` because `RwLock` is way slower

Reply:
Using Mutex would prevent us from making `cpu_count` cheap

Reply:
for a cheap `cpu_count()` impl, we should just store that in an atomicusize, like we already do on x86.

Regardless of perf, the question is mostly about semantics -- do we actually need multiple readers to be able to immutably access the vector of online `CpuId`s? If so, why, and in what circumstances? (I'm not saying there aren't any, I just want to explore this)

Reply:
> Regardless of perf, my question is primarily about semantics -- do we actually _need_ multiple readers to be able to immutably access the vector of online `CpuId`s? If so, why, and in what circumstances? (I'm not saying there aren't any, I just want to explore this)

The only use case I have in mind really is for `cpu_count`, so an atomicusize solves the issue.